### PR TITLE
docs: updating Optimism's org structure

### PR DESCRIPTION
**Pull request by sbvegan**
**Description**

From what I understand the Optimism organization moved from OptimismPBC to Optimism Foundation and OP Labs. I updated the `specs` `README` doc to reflect that.

**Tests**

n/a - documentation 

**Invariants**

n/a - documentation

**Additional context**

n/a

### CODE REVIEW

**Review by maurelian — APPROVED** [review]
LGTM thank you

File: specs/README.md [hunk 1]
```diff
@@ -44,5 +44,5 @@ Our aim is to design a protocol specification that is:
 - **Secure:** This is self-evident. We cannot lose money and in a system where even a bit of
   downtime can result in loss of funds this means everything we build must be incredibly secure.
 - **Decentralizable:** Everything we build must have a clear path towards decentralization. Today
-  Optimism relies on OptimismPBC to function, but eventually it will be managed by a DAO and even in
+  Optimism relies on the Optimism Foundation and OP Labs to function, but eventually it will be managed by a DAO and even in
```
Comment (maurelian) [inline specs/README.md:47]: CI is failing because this line is longer than 100 characters. Fix that and this should be good to go.

**Comment by sbvegan** [comment]
@mergifyio requeue

**Comment by sbvegan** [comment]
@maurelian I fixed the line length but it looks like we have a failing flaky test

**Review by tynes — DISMISSED** [review]
Explicitly requesting changes until https://github.com/ethereum-optimism/optimism/pull/4629/files#r1064835005 is addressed

File: specs/README.md [hunk 2]
```diff
@@ -44,5 +44,5 @@ Our aim is to design a protocol specification that is:
 - **Secure:** This is self-evident. We cannot lose money and in a system where even a bit of
   downtime can result in loss of funds this means everything we build must be incredibly secure.
 - **Decentralizable:** Everything we build must have a clear path towards decentralization. Today
```
[the remark this answers is not in the document]
Comment (sbvegan) [reply specs/README.md]: Sounds good, it has been updated!

**Comment by sbvegan** [comment]
@tynes I updated that statement, let me know if it needs any other changes

**Review by roninjin10 — APPROVED** [review]

**Review by protolambda — APPROVED** [review]

**Comment by protolambda** [comment]
@Mergifyio rebase

**Pull request closed without merging**